Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -366,9 +366,22 @@ namespace AzToolsFramework
                 {
                     priorColumnIndex = m_columnLayout->indexOf(m_domOrderedChildren[searchIndex]);
                 }
+
+                // if the alignment attribute is present, insert the widget with the appropriate alignment
+                auto alignment = AZ::Dpe::Nodes::PropertyEditor::Alignment.ExtractFromDomNode(childValue);
+                if (alignment.has_value())
+                {
+                    (alignment.value() == AZ::Dpe::Nodes::PropertyEditor::Align::AlignRight
+                        ? m_columnLayout->insertWidget(priorColumnIndex + 1, addedWidget, 0, Qt::AlignRight)
+                        : m_columnLayout->insertWidget(priorColumnIndex + 1, addedWidget, 0, Qt::AlignLeft));
```

Comment:
Might be easier in the future to add additional alignment types if you do something like:

```
Qt::Alignment widgetAlignment;
switch (alignment.value())
{
case AZ::Dpe::Nodes::PropertyEditor::Align::AlignLeft:
    widgetAlignment = Qt::AlignLeft;
case AZ::Dpe::Nodes::PropertyEditor::Align::AlignRight:
    widgetAlignment = Qt::AlignRight;
}

m_columnLayout->insertWidget(priorColumnIndex + 1, addedWidget, 0, widgetAlignment);
```

Reply:
Thanks, updated!